### ISSUE TITLE
`fish_pager_color_secondary` should color the entire line

### ISSUE THREAD

**Issue by ariasuni** [fish-shell/fish-shell#4866]
When I set `fish_pager_color_secondary` to a color background, I see those two problems:

When option doesn’t have a description, the background color covers only the option’s name:
![screenshot_20180330_190024](https://user-images.githubusercontent.com/2446451/38146082-ba6cbebe-344c-11e8-9b29-f1ee8b5aa3d4.png)

When there is a long and a short option, there’s a gap in-between.
![screenshot_20180330_190107](https://user-images.githubusercontent.com/2446451/38146083-ba8969a6-344c-11e8-97e7-18862106d37c.png)

This boils down to: fish should color the entire «line».

**Comment by faho** [comment]
I just implemented this. I removed both the gap after a candidate with no description (the "--abbrev" case) and the gap between candidates with the same meaning (the "--color/--colour" or long/short case).

So this changes it from this

![screenshot_20180331_124928](https://user-images.githubusercontent.com/5185367/38162593-ec697dc0-34e3-11e8-9cdd-ae3c4ddd5430.png)

to this

![screenshot_20180331_124905](https://user-images.githubusercontent.com/5185367/38162594-f0d3d856-34e3-11e8-9b4c-c7c4e7a32d68.png)

which I feel looks much nicer.
